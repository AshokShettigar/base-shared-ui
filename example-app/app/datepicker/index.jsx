import { connect } from 'react-redux';
import { Button, CheckBox, DatePicker, Input } from 'platform-common-ui';
import BaseComponentView from 'app/components/base-component-view';

const ConnectedDatePicker = connect()(DatePicker);
export default class DatePickerExample extends BaseComponentView {
    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
            hideLabel: false,
            isDisabled: false,
            isRequired: false,
            label: 'Date picker',
            placeholder: 'mm/dd/yyyy',
            value: null
        };

        this.onChange = value => this.setState({ value });
        this.onDisabledChange = () => this.setState({ isDisabled: !this.state.isDisabled });
        this.onRequiredChange = () => this.setState({ isRequired: !this.state.isRequired });
        this.onHasErrorChange = () => this.setState({ hasError: !this.state.hasError });
        this.onHideLabelChange = () => this.setState({ hideLabel: !this.state.hideLabel });
        this.onChangePlaceholder = placeholder => this.setState({ placeholder });
        this.onChangeLabel = label => this.setState({ label });
        this.setToToday = () => this.setState({ value: new Date() });
        this.clearValue = () => this.setState({ value: null });
    }

    getOptions() {
        return (
            <div>
                <Button onClick={this.setToToday}>
                    Set to Today
                </Button>
                <Button onClick={this.clearValue}>
                    Clear date
                </Button>
                <CheckBox
                    id="fieldHideLabel"
                    isChecked={this.state.hideLabel}
                    onChange={this.onHideLabelChange}
                    value="hideLabel"
                >
                    Hide Label
                </CheckBox>
                <CheckBox
                    id="fieldDisable"
                    isChecked={this.state.isDisabled}
                    onChange={this.onDisabledChange}
                    value="disable"
                >
                    Disable
                </CheckBox>
                <CheckBox
                    id="fieldRequired"
                    isChecked={this.state.isRequired}
                    onChange={this.onRequiredChange}
                    value="required"
                >
                    Enable Required
                </CheckBox>
                <CheckBox
                    id="fieldHasError"
                    isChecked={this.state.hasError}
                    onChange={this.onHasErrorChange}
                    value="error"
                >
                    Set error
                </CheckBox>
                <Input
                    id="placeholderInput"
                    label="Placeholder value"
                    onChange={this.onChangePlaceholder}
                    value={this.state.placeholder}
                />
                <Input
                    id="labelInput"
                    label="Label value"
                    onChange={this.onChangeLabel}
                    value={this.state.label}
                />
            </div>
        );
    }

    renderComponent() {
        return (
            <ConnectedDatePicker
                hasError={this.state.hasError}
                hideLabel={this.state.hideLabel}
                id="dp1"
                isDisabled={this.state.isDisabled}
                isRequired={this.state.isRequired}
                label={this.state.label}
                onChange={this.onChange}
                placeholder={this.state.placeholder}
                value={this.state.value}
            />
        );
    }
}
