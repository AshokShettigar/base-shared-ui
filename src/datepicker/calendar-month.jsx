import { Component } from 'react';
import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';
import { Icon } from 'platform-common-ui';

const DURATION = 150;
const MONTHS = [
    'January',
    'Feburary',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December'
];

export function compareYearAndMonth(year, month, prevYear, prevMonth) {
    if (year === prevYear && month === prevMonth) { return 0; }
    if (year === prevYear && month > prevMonth) { return 1; }
    if (year > prevYear) { return 1; }
    return -1;
}

export default class CalendarMonth extends Component {
    constructor(props) {
        super(props);
        this.state = {
            transitionName: 'month-next'
        };

        this.onMouseDown = event => event.preventDefault();
        this.onNavigateToNextMonth = () =>
            this.updateYearAndMonth(this.props.year, this.props.month + 1);
        this.onNavigateToPreviousMonth = () =>
            this.updateYearAndMonth(this.props.year, this.props.month - 1);
    }

    componentWillReceiveProps(nextProps) {
        const comparedValue = compareYearAndMonth(
            nextProps.year,
            nextProps.month,
            this.props.year,
            this.props.month);
        if (comparedValue !== 0) {
            this.setState({
                transitionName: comparedValue > 0 ? 'month-next' : 'month-prev'
            });
        }
    }

    updateYearAndMonth(year, month) {
        const date = new Date(year, month);
        this.props.onChangeMonthAndYear(date.getMonth(), date.getFullYear());
    }

    render() {
        return (
            // eslint-disable-next-line jsx-a11y/interactive-supports-focus
            <div
                className="cn-calendar-month"
                onMouseDown={this.onMouseDown}
                role="button"
            >
                <Icon
                    className="cn-calendar-month__nav-month-btn"
                    icon="triangle-left"
                    onClick={this.onNavigateToPreviousMonth}
                />
                <div className="cn-calendar-month__animation-container">
                    <CSSTransitionGroup
                        className="cn-calendar-month__animation"
                        transitionName={this.state.transitionName}
                        transitionEnterTimeout={DURATION}
                        transitionLeaveTimeout={DURATION}
                    >
                        <div
                            className="cn-calendar-month__text"
                            key={`${this.props.year}_${this.props.month}`}
                        >
                            {`${MONTHS[this.props.month]} ${this.props.year}`}
                        </div>
                    </CSSTransitionGroup>
                </div>
                <Icon
                    className="cn-calendar-month__nav-month-btn"
                    icon="triangle-right"
                    onClick={this.onNavigateToNextMonth}
                />
            </div>
        );
    }
}

CalendarMonth.propTypes = {
    month: PropTypes.number.isRequired,
    onChangeMonthAndYear: PropTypes.func.isRequired,
    year: PropTypes.number.isRequired
};
