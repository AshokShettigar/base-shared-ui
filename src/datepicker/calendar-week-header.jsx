const DAYS_OF_WEEK = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];

export default () => (
    <div className="cn-calendar-week-header">
        {DAYS_OF_WEEK.map(day => (
            <div className="cn-calendar__column" key={day}>
                {day}
            </div>
        ))}
    </div>
);
