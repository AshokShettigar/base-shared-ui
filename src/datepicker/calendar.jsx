import PropTypes from 'prop-types';
import { Popup } from 'react-redux-popup';
import CalendarBody from 'platform-common-ui/datepicker/calendar-body';
import CalendarMonth from 'platform-common-ui/datepicker/calendar-month';
import CalendarWeekHeader from 'platform-common-ui/datepicker/calendar-week-header';

const updateSelectedDateHandler =
    (year, month, onChange) => day => () => onChange(new Date(year, month, day));

export const Calendar = props => (
    <div className="cn-calendar">
        <CalendarMonth
            month={props.month}
            onChangeMonthAndYear={props.onChangeMonthAndYear}
            year={props.year}
        />
        <CalendarWeekHeader />
        <CalendarBody
            highlightedDate={props.highlightedDate}
            month={props.month}
            onUpdateSelectedDate={
                updateSelectedDateHandler(props.year, props.month, props.onChange)
            }
            selectedDate={props.selectedDate}
            year={props.year}
        />
    </div>
);

Calendar.propTypes = {
    highlightedDate: PropTypes.instanceOf(Date).isRequired,
    month: PropTypes.number.isRequired,
    onChange: PropTypes.func.isRequired,
    onChangeMonthAndYear: PropTypes.func.isRequired,
    selectedDate: PropTypes.instanceOf(Date),
    year: PropTypes.number.isRequired
};

export default Popup(Calendar);
