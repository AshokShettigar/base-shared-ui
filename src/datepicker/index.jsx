import classNames from 'classnames';
import { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { closePopup, openPopup } from 'react-redux-popup';
import CalendarPopup from 'platform-common-ui/datepicker/calendar';
import { Icon } from 'platform-common-ui';
import { Input } from 'platform-common-ui/input';

const VALID_DATE_REGEX = /^(\d|\d{2})\/(\d|\d{2})\/(\d{4})$/;
const ARROW_KEY_CODE = {
    37: -1,
    38: -7,
    39: 1,
    40: 7
};

export function padLeft(value) {
    if (value > 9) { return value; }
    return `0${value}`;
}

export function formatDate(date) {
    const month = date.getMonth() + 1;
    const year = date.getFullYear();
    const day = date.getDate();
    return `${padLeft(month)}/${padLeft(day)}/${year}`;
}

export function isValidDate(value) {
    return VALID_DATE_REGEX.test(value) && !isNaN(Date.parse(value));
}

export class DatePicker extends Component {
    constructor(props) {
        super(props);
        this.state = this.getInitialState(this.props.value);
        this.getRect = () => this.inputElem.textInput.getBoundingClientRect();
        this.onChangeMonthAndYear = this.onChangeMonthAndYearHandler.bind(this);
        this.onChangeSelectedDate = this.onChangeSelectedDateHandler.bind(this);
        this.onClosePopup = this.onClosePopupHandler.bind(this);
        this.onKeyDown = this.onKeyDownHandler.bind(this);
        this.onValueChange = this.onValueChangeHandler.bind(this);
        this.setDatePickerElemRef = component => { this.datePickerElem = component; };
        this.setInputElemRef = component => { this.inputElem = component; };
    }

    getInitialState(selectedDate) {
        const formattedDate = selectedDate ? formatDate(selectedDate) : '';
        const highlightedDate = selectedDate || new Date();
        const month = highlightedDate.getMonth();
        const year = highlightedDate.getFullYear();
        return {
            formattedDate,
            highlightedDate,
            month,
            selectedDate,
            year,
            isPopupOpen: false
        };
    }

    componentDidUpdate() {
        // this condition is to take care of a case when the property value is changed
        // outside of this component
        if (this.props.value !== this.state.selectedDate) {
            // unfortunately, this isn't possible because when componentWillReceiveProps is
            // called, the state is not yet updated
            // eslint-disable-next-line react/no-did-update-set-state
            this.setState(this.getInitialState(this.props.value));
        }
    }

    onClickHandler(id) {
        return () => {
            if (this.props.isDisabled) { return; }
            this.setState({ isPopupOpen: true });
            this.props.openPopup(id);
        };
    }

    onKeyDownHandler(keyCode) {
        if (!this.state.isPopupOpen) { return; }

        if (keyCode > 36 && keyCode < 41) {
            const highlightedDate = new Date(
                this.state.highlightedDate.getFullYear(),
                this.state.highlightedDate.getMonth(),
                this.state.highlightedDate.getDate() + ARROW_KEY_CODE[keyCode]);
            this.setState({
                highlightedDate,
                month: highlightedDate.getMonth(),
                year: highlightedDate.getFullYear()
            });
        } else if (keyCode === 13) {
            this.inputElem.blur();
            this.onChangeSelectedDate(this.state.highlightedDate);
        } else if (keyCode === 27) {
            this.inputElem.blur();
            this.onClosePopupHandler();
        }
    }

    onChangeMonthAndYearHandler(month, year) {
        this.setState({ month, year });
        this.inputElem.focus();
    }

    onChangeSelectedDateHandler(selectedDate) {
        const formattedDate = formatDate(selectedDate);
        const month = selectedDate.getMonth();
        const year = selectedDate.getFullYear();
        this.setState({
            formattedDate,
            month,
            selectedDate,
            year,
            highlightedDate: selectedDate,
            isPopupOpen: false
        });
        this.props.onChange(selectedDate);
        this.props.closePopup(this.props.id);
    }

    onClosePopupHandler() {
        const formattedDate = this.state.selectedDate ? formatDate(this.state.selectedDate) : '';
        const highlightedDate = this.state.selectedDate || new Date();
        const month = highlightedDate.getMonth();
        const year = highlightedDate.getFullYear();
        this.setState({
            formattedDate,
            highlightedDate,
            month,
            year,
            isPopupOpen: false
        });
        this.props.closePopup(this.props.id);
    }

    onValueChangeHandler(value) {
        if (value && !isValidDate(value)) {
            this.setState({ formattedDate: value });
            return;
        }

        const selectedDate = value ? new Date(value) : null;
        const highlightedDate = selectedDate || new Date();
        const month = highlightedDate.getMonth();
        const year = highlightedDate.getFullYear();
        this.setState({
            highlightedDate,
            month,
            selectedDate,
            year,
            formattedDate: value
        });
        this.props.onChange(selectedDate);
    }

    stopEvent(event) {
        event.stopPropagation();
    }

    render() {
        const className = classNames('cn-date-picker', this.props.className);
        const iconClassName = classNames({
            'cn-date-picker__icon': true,
            'cn-date-picker__icon--is-active': this.state.isPopupOpen
        });

        return (
            // eslint-disable-next-line jsx-a11y/no-static-element-interactions
            <div
                className={className}
                onClick={this.onClickHandler(this.props.id)}
                onMouseUp={this.stopEvent}
                ref={this.setDatePickerElemRef}
            >
                <Input
                    className="cn-date-picker__input"
                    hasError={this.props.hasError}
                    hideLabel={this.props.hideLabel}
                    id={this.props.id}
                    isDisabled={this.props.isDisabled}
                    isRequired={this.props.isRequired}
                    label={this.props.label}
                    labelClassName="cn-date-picker__label"
                    onChange={this.onValueChange}
                    onKeyDown={this.onKeyDown}
                    placeholder={this.props.placeholder}
                    ref={this.setInputElemRef}
                    value={this.state.formattedDate}
                >
                    <Icon className={iconClassName} icon="calendar" />
                </Input>
                <CalendarPopup
                    closePopup={this.onClosePopup}
                    getRect={this.getRect}
                    highlightedDate={this.state.highlightedDate}
                    id={this.props.id}
                    month={this.state.month}
                    popupClassName="cn-date-picker__popup"
                    offset={3}
                    onChange={this.onChangeSelectedDate}
                    onChangeMonthAndYear={this.onChangeMonthAndYear}
                    selectedDate={this.state.selectedDate}
                    year={this.state.year}
                />
            </div>
        );
    }
}

DatePicker.propTypes = {
    className: PropTypes.string,
    closePopup: PropTypes.func.isRequired,
    hasError: PropTypes.bool.isRequired,
    hideLabel: PropTypes.bool.isRequired,
    id: PropTypes.string.isRequired,
    isDisabled: PropTypes.bool.isRequired,
    isRequired: PropTypes.bool.isRequired,
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    openPopup: PropTypes.func.isRequired,
    placeholder: PropTypes.string.isRequired,
    value: PropTypes.instanceOf(Date)
};

DatePicker.defaultProps = {
    hasError: false,
    hideLabel: false,
    isDisabled: false,
    isRequired: false,
    placeholder: 'mm/dd/yyyy'
};

export const ConnectedDatePicker = connect(null, { closePopup, openPopup })(DatePicker);
