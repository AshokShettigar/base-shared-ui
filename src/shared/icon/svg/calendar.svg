<?xml version="1.0" encoding="utf-8"?>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20">
    <title>calendar</title>
    <path d="M18.5,19.1H1.5a.6.6,0,0,1-.6-.6V2.7a.6.6,0,0,1,.6-.6H4.4a.6.6,0,1,1,0,1.2H2.1V17.9H17.9V3.3H15.6a.6.6,0,1,1,0-1.2h2.9a.6.6,0,0,1,.6.6V18.5A.6.6,0,0,1,18.5,19.1Z"/>
    <path d="M6.8,5H4.4a.6.6,0,0,1-.6-.6V1.5A.6.6,0,0,1,4.4.9H6.8a.6.6,0,0,1,.6.6V4.4A.6.6,0,0,1,6.8,5ZM5,3.8H6.2V2.1H5Z"/>
    <path d="M15.6,5H13.2a.6.6,0,0,1-.6-.6V1.5a.6.6,0,0,1,.6-.6h2.3a.6.6,0,0,1,.6.6V4.4A.6.6,0,0,1,15.6,5ZM13.8,3.8H15V2.1H13.8Z"/>
    <path d="M13.2,3.3H6.8a.6.6,0,1,1,0-1.2h6.4a.6.6,0,1,1,0,1.2Z"/>
    <path d="M18.5,7.4H1.5a.6.6,0,1,1,0-1.2h17a.6.6,0,1,1,0,1.2Z"/>
    <path d="M6.2,16.8a.4.4,0,0,1-.4-.4V8.8a.4.4,0,1,1,.9,0v7.6A.4.4,0,0,1,6.2,16.8Z"/>
    <path d="M9.7,16.8a.4.4,0,0,1-.4-.4V8.8a.4.4,0,1,1,.9,0v7.6A.4.4,0,0,1,9.7,16.8Z"/>
    <path d="M13.3,16.8a.4.4,0,0,1-.4-.4V8.8a.4.4,0,0,1,.9,0v7.6A.4.4,0,0,1,13.3,16.8Z"/>
    <path d="M16,10.6H4a.4.4,0,0,1,0-.9H16a.4.4,0,0,1,0,.9Z"/>
    <path d="M16,13.1H4a.4.4,0,0,1,0-.9H16a.4.4,0,0,1,0,.9Z"/>
    <path d="M16,15.6H4a.4.4,0,0,1,0-.9H16a.4.4,0,0,1,0,.9Z"/>
</svg>
