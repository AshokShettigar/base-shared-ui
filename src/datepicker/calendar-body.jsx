import classNames from 'classnames';
import { Component } from 'react';
import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';
import { compareYearAndMonth } from 'platform-common-ui/datepicker/calendar-month';

const DURATION = 150;

export function getDaysInMonth(year, month) {
    if (month < 7) {
        if (month === 1) {
            return year % 4 === 0 ? 29 : 28;
        }
        return month % 2 === 0 ? 31 : 30;
    }
    return month % 2 === 0 ? 30 : 31;
}

export function renderDay(key, day, isSelected, isHighlighted, isDisabled, onUpdateSelectedDate) {
    const className = classNames({
        'cn-calendar__column': true,
        'cn-calendar__day': true,
        'cn-calendar__day--is-disabled': isDisabled,
        'cn-calendar__day--is-highlighted': isHighlighted,
        'cn-calendar__day--is-selected': isSelected
    });
    const onClick = isDisabled ? null : onUpdateSelectedDate(day);
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    return <div className={className} key={key} onClick={onClick}>{day}</div>;
}

export function renderPreviousMonth(currentYear, currentMonth, currentMonthStartDayOfWeek) {
    // fill out the dates from previous month before the 1st starts in the current month
    const prevMonth = new Date(currentYear, currentMonth - 1);
    const prevDaysInMonth = getDaysInMonth(prevMonth.getYear(), prevMonth.getMonth());
    const days = [];
    for (let i = currentMonthStartDayOfWeek - 1; i > -1; i--) {
        days.push(renderDay(`s_${i}`, prevDaysInMonth - i, false, false, true));
    }
    return days;
}

export function renderCurrentMonth(
    year,
    month,
    daysInMonth,
    selectedDate,
    highlightedDate,
    onUpdateSelectedDate
) {
    const selectedDay = selectedDate && selectedDate.getDate();
    const selectedMonth = selectedDate && selectedDate.getMonth();
    const selectedYear = selectedDate && selectedDate.getFullYear();
    const highlightedDay = highlightedDate.getDate();
    const highlightedMonth = highlightedDate.getMonth();
    const highlightedYear = highlightedDate.getFullYear();
    const days = [];
    for (let i = 0; i < daysInMonth; i++) {
        const day = i + 1;
        const isSelected = year === selectedYear
            && month === selectedMonth
            && day === selectedDay;
        const isHighlighted = year === highlightedYear
            && month === highlightedMonth
            && day === highlightedDay;
        days.push(renderDay(
            `d_${day}`,
            day,
            isSelected,
            isHighlighted,
            false,
            onUpdateSelectedDate
        ));
    }
    return days;
}

export function renderNextMonth(startDayOfWeek, daysInMonth) {
    // fill in the rest of the calendar block with date from next month
    const fillDaysNum = 7 - ((startDayOfWeek + daysInMonth) % 7);
    const days = [];
    if (fillDaysNum < 7) {
        for (let r = 0; r < fillDaysNum; r++) {
            days.push(renderDay(`r_${r}`, r + 1, false, false, true));
        }
    }
    return days;
}

export default class CalendarBody extends Component {
    constructor(props) {
        super(props);
        this.state = {
            height: 120,
            transitionName: 'month-next'
        };
    }

    componentDidMount() {
        // eslint-disable-next-line react/no-did-mount-set-state
        this.setState({ height: this.elem.clientHeight });
    }

    componentWillReceiveProps(nextProps) {
        const comparedValue = compareYearAndMonth(
            nextProps.year,
            nextProps.month,
            this.props.year,
            this.props.month);
        if (comparedValue !== 0) {
            this.setState({
                transitionName: comparedValue > 0 ? 'month-next' : 'month-prev'
            });
        }
    }

    componentDidUpdate(prevProps) {
        if (prevProps.year !== this.props.year || prevProps.month !== this.props.month) {
            // eslint-disable-next-line react/no-did-update-set-state
            this.setState({ height: this.elem.clientHeight });
        }
    }

    render() {
        const {
            highlightedDate,
            month,
            onUpdateSelectedDate,
            selectedDate,
            year
        } = this.props;

        const date = new Date(year, month);
        const startDayOfWeek = date.getDay();
        const daysInMonth = getDaysInMonth(year, month);

        return (
            <div className="cn-calendar-body" style={{ height: this.state.height }}>
                <CSSTransitionGroup
                    transitionName={this.state.transitionName}
                    transitionEnterTimeout={DURATION}
                    transitionLeaveTimeout={DURATION}
                >
                    <div
                        className="cn-calendar-body__month"
                        key={`${year}_${month}`}
                        ref={c => { this.elem = c; }}
                    >
                        {renderPreviousMonth(year, month, startDayOfWeek)}
                        {renderCurrentMonth(
                            year,
                            month,
                            daysInMonth,
                            selectedDate,
                            highlightedDate,
                            onUpdateSelectedDate)}
                        {renderNextMonth(startDayOfWeek, daysInMonth)}
                    </div>
                </CSSTransitionGroup>
            </div>
        );
    }
}

CalendarBody.propTypes = {
    highlightedDate: PropTypes.instanceOf(Date).isRequired,
    month: PropTypes.number.isRequired,
    onUpdateSelectedDate: PropTypes.func.isRequired,
    selectedDate: PropTypes.instanceOf(Date),
    year: PropTypes.number.isRequired
};
