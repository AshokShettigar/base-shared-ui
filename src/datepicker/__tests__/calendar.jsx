import { mount, shallow } from 'enzyme';
import { Calendar } from 'platform-common-ui/datepicker/calendar';

describe('datepicker/calendar', function() {
    it('should render the component', function() {
        const onChange = () => true;
        const onChangeMonthAndYear = () => true;
        const highlightedDate = new Date(Date.UTC(2017, 0, 1));
        const wrapper = shallow(
            <Calendar
                highlightedDate={highlightedDate}
                month={0}
                onChange={onChange}
                onChangeMonthAndYear={onChangeMonthAndYear}
                year={2017}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onChange callback on selected date change', function(done) {
        const onChange = selectedDate => {
            expect(selectedDate.getFullYear()).toBe(2017);
            expect(selectedDate.getMonth()).toBe(0);
            expect(selectedDate.getDate()).toBe(1);
            done();
        };
        const onChangeMonthAndYear = () => true;
        const highlightedDate = new Date(2017, 0, 1);
        const wrapper = mount(
            <Calendar
                highlightedDate={highlightedDate}
                month={0}
                onChange={onChange}
                onChangeMonthAndYear={onChangeMonthAndYear}
                year={2017}
            />
        );
        wrapper.find('.cn-calendar__day--is-highlighted').simulate('click');
    });
});
