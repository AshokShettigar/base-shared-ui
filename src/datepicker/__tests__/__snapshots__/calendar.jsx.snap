// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datepicker/calendar should render the component 1`] = `
<div
  className="cn-calendar"
  data-component="Calendar"
>
  <CalendarMonth
    month={0}
    onChangeMonthAndYear={[Function]}
    year={2017}
  />
  <Component />
  <CalendarBody
    highlightedDate={2017-01-01T00:00:00.000Z}
    month={0}
    onUpdateSelectedDate={[Function]}
    year={2017}
  />
</div>
`;
